### PR TITLE
elk: 7.5.1 -> 7.11.1

### PR DESCRIPTION
**Pull request by thefloweringash**
###### Motivation for this change

Package update and support for aarch64-linux, but unfortunately not a simple update. This requires touching a lot of related packages, most of which I'm not very familiar with. Hopefully I can at least I can get the ball rolling on the update.

Notably:

- Drops elasticsearch7-oss and kibana7-oss, since they're no longer distributed by upstream as of 7.11. We might be able to provide these under the SSPL, but they wouldn't fit into the current pull request.
- ~Breaks `search-guard` and `elasticsearch-lemmagen` plugins, which don't seem support elasticsearch 7.11.x yet.~ Plugins updated!

cc maintainers: @apeschar, @basvandijk, @wjlroe, @offlinehacker

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS (aarch64-linux and x86_64-linux)
   - [x] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/search/elasticsearch/7.x.nix [hunk 1]
```diff
@@ -46,38 +41,43 @@ stdenv.mkDerivation (rec {
...
       --prefix PATH : "${makeBinPath [ util-linux coreutils gnugrep ]}" \
       --set JAVA_HOME "${jre_headless}"
 
     wrapProgram $out/bin/elasticsearch-plugin --set JAVA_HOME "${jre_headless}"
   '';
 
-  passthru = { inherit enableUnfree; };
+  postFixup = lib.optionalString stdenv.hostPlatform.isLinux ''
+    for exe in $(find $out/modules/x-pack-ml/platform/${plat}-${arch}/bin -executable -type f); do
+      echo "patching $exe..."
```
Comment (SuperSandro2000) [inline pkgs/servers/search/elasticsearch/7.x.nix:70]: Doesn't patchelf echo what it is doing?

Comment (thefloweringash) [reply pkgs/servers/search/elasticsearch/7.x.nix]: I don't see what this `postFixup` hook is doing that `autoPatchelfHook` doesn't, so I'm now removing it.

**Comment by thefloweringash** [comment]
I've marked this as ready for review, but would appreciate additional testing and the maintainer's input here.

File: pkgs/misc/logging/beats/7.x.nix [hunk 2]
```diff
@@ -1,17 +1,17 @@
-{ lib, fetchFromGitHub, elk7Version, buildGoPackage, libpcap, systemd }:
+{ lib, fetchFromGitHub, elk7Version, buildGoModule, libpcap, systemd }:
 
-let beat = package : extraArgs : buildGoPackage (rec {
+let beat = package : extraArgs : buildGoModule (rec {
       name = "${package}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/misc/logging/beats/7.x.nix:4]: That could be converted to pname + version.

File: pkgs/servers/search/elasticsearch/7.x.nix [hunk 3]
```diff
@@ -15,19 +14,15 @@ let
...
+      x86_64-linux  = "1ld7656b37l67vi4pyv0il865b168niqnbd4hzbvdnwrm35prp10";
+      x86_64-darwin = "11b180y11xw5q01l7aw6lyn15lp9ks8xmakjg1j7gp3z6c90hpn3";
+      aarch64-linux = "0s4ph79x17f90jk31wjwk259dk9dmhnmnkxdcn77m191wvf6m3wy";
     };
 in
-stdenv.mkDerivation (rec {
+stdenv.mkDerivation rec {
   version = elk7Version;
-  name = "elasticsearch-${optionalString (!enableUnfree) "oss-"}${version}";
+  name = "elasticsearch-${version}";
```
Comment (SuperSandro2000) [inline pkgs/servers/search/elasticsearch/7.x.nix:25]:
```suggestion
  pname = "elasticsearch";
```

**Comment by thefloweringash** [comment]
Pushed an update to use `pname` everywhere I saw `name`.

I've been using this expression to keep a track of the packages changed and sources this update requires: [gist](https://gist.github.com/thefloweringash/8d359cadb8f3c8c32f3d96c896ac9b3a).

**Comment by zbioe** [comment]
Thank's for this PR. I really appreciate it

**Comment by thefloweringash** [comment]
I've updated this to fix merge conflicts only.

Since I opened this the versions have changed. The current version elasticsearch is 7.13, but some plugins are limited (searchguard goes up to 7.12.1, lemmagen goes up to 7.12). I don't know what's best here: to hold back elasticsearch, or to keep the maximum plugin compatibility.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -4367,11 +4367,6 @@ in
     util-linux = util-linuxMinimal;
     jre_headless = jre8_headless; # TODO: remove override https://github.com/NixOS/nixpkgs/pull/89731
   };
-  elasticsearch7-oss = callPackage ../servers/search/elasticsearch/7.x.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:4370]: We need aliases entries for the removals.

**Comment by doronbehar** [comment]
@thefloweringash it seems you have addressed all review comments, there's a small merge conflict now. 

As for:

> but some plugins are limited (searchguard goes up to 7.12.1, lemmagen goes up to 7.12). I don't know what's best here: to hold back elasticsearch, or to keep the maximum plugin compatibility.

It seems not too many plugins are incompatible. I think you should update elasticsearch to the latest, and mark these plugins as broken for elasticsearch 7.3, as done here:

https://github.com/NixOS/nixpkgs/blob/3d72b0b6b0ff8a4cd46ebad0382d8d06d7255e95/pkgs/servers/search/elasticsearch/plugins.nix#L38-L41

**Comment by happysalada** [comment]
I'm only seeing this PR now.
I've made the upgrade to elk 7.10.2 which is the last version including oss.
Most of the fixes in this PR are integrated, except the hashes for version 7.11.1 of course.
There is one more weird fix needed to get elasticsearch working properly, the jvm gc.log option needs to be a writeable path.
I've got the fix in this PR https://github.com/NixOS/nixpkgs/pull/136255
There are 2 things that I'm not sure about. If anybody is interested, it could use some feedback.

**Comment by roberth** [comment]
I've rebased and successfully tested `nixosTests.elk.unfree`.

**Comment by roberth** [comment]
I've rebased and successfully tested `nixosTests.elk.unfree`.

**Review by roberth — APPROVED** [review]

**Pull request merged**